catalyst_voices/apps/voices/lib/widgets/text_field/date_picker/voices_date_picker_field.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:catalyst_voices/widgets/text_field/date_picker/date_picker_controller.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+part 'base_picker.dart';
+part 'voices_calendar_picker.dart';
+part 'voices_time_picker.dart';
+
+enum DateTimePickerType { date, time }
+
+final GlobalKey calendarKey = GlobalKey();
+final GlobalKey timeKey = GlobalKey();
+
+class ScrollControllerProvider extends InheritedWidget {
+  final ScrollController scrollController;
+
+  const ScrollControllerProvider({
+    super.key,
+    required this.scrollController,
+    required super.child,
+  });
+
+  static ScrollController of(BuildContext context) {
+    final provider =
+        context.dependOnInheritedWidgetOfExactType<ScrollControllerProvider>();
+    return provider!.scrollController;
+  }
+
+  static ScrollController? maybeOf(BuildContext context) {
+    final provider =
+        context.dependOnInheritedWidgetOfExactType<ScrollControllerProvider>();
+    return provider?.scrollController;
+  }
+
+  @override
+  bool updateShouldNotify(ScrollControllerProvider oldWidget) {
+    return scrollController != oldWidget.scrollController;
+  }
+}
+
+class VoicesDatePicker extends StatefulWidget {
+  final DatePickerController controller;
+  final String timeZone;
+  const VoicesDatePicker({
```

Comment:
Idea for moving / renaming classes.

Let's say we have `VoicesDateTimeTextField` which have `ValueChanged<DateTime> onChanged` and simple controller(`VoicesDateTimeController`) which maybe extends `TextEditingController` or simply `ValueNotifier<DateTime>`. 

Goal here is for `VoicesDateTimeTextField` to output only `DateTime` and accept it as input for `VoicesDateTimeController`.


This widget builds two childs in a row (will be more complex with decoration), `VoicesDateTextField` and `VoicesTimeOfDayTextField` both of which are stateless and accept `TextEditingController` from `VoicesDateTimeTextField` and have `onCalendarTap` / `onClockTap` respectively. 

Now `VoicesDateTimeTextField` can launch correct picker and global keys can be private. 
Ofc. `VoicesDateTimeTextField` is watching changes in both fields and merging them into final `DateTime`.

Reply:
It's good idea, but in this case, we have two possibilities about from where data come from: either textfield or picker. We also need to validate this data. If we want to separate controllers then your approach is better and I'm happy to do it.

Reply:
3 controllers even. I think it will be easier to read and maintain

`VoicesDateTimeTextField` will have to open `OverlayEntry` and update respective controller, yes, and `VoicesDateTextField` / `VoicesTimeOfDayTextField` will have to watch changes in its controller and validate them but it's happening automatically i believe.

Pseudo code
```dart
// voices_date_time_text_field.dart

class VoicesDateTimeTextFieldController extends ValueNotifier<DateTime?> {
  VoicesDateTimeTextFieldController([super._value]);
}

class VoicesDateTimeTextField extends StatefulWidget {
  final VoicesDateTimeTextFieldController? controller;
  final ValueChanged<DateTime?>? onChanged;

  const VoicesDateTimeTextField({
    super.key,
    this.controller,
    this.onChanged,
  });

  @override
  State<VoicesDateTimeTextField> createState() {
    return _VoicesDateTimeTextFieldState();
  }
}

class _VoicesDateTimeTextFieldState extends State<VoicesDateTimeTextField> {
  late final VoicesDateTextFieldController _dateController;
  late final VoicesTimeOfDayTextFieldController _timeOfDayController;

  @override
  void initState() {
    super.initState();

    _dateController = VoicesDateTextFieldController()..addListener(_xxx);
    _timeOfDayController = VoicesTimeOfDayTextFieldController()
      ..addListener(_xxx);
  }

  @override
  Widget build(BuildContext context) {
    return Row(
      children: [],
    );
  }

  void _xxx() {
    //
  }
}

// voices_date_text_field.dart

class VoicesDateTextFieldController extends ValueNotifier<DateTime?> {
  VoicesDateTextFieldController([super._value]);
}

class VoicesDateTextField extends StatefulWidget {
  final VoicesDateTextFieldController controller;

  const VoicesDateTextField({
    super.key,
    required this.controller,
  });

  @override
  State<VoicesDateTextField> createState() => _VoicesDateTextFieldState();
}

class _VoicesDateTextFieldState extends State<VoicesDateTextField> {
  late final TextEditingController _textEditingController;

  @override
  void initState() {
    super.initState();
    _textEditingController = TextEditingController();
  }

  @override
  Widget build(BuildContext context) {
    return VoicesTextField(
      controller: _textEditingController,
      onFieldSubmitted: (value) {},
    );
  }
  
  // listen _textEditingController, validate, parse and pass result to widget.controller
}

// voices_time_of_day_text_field.dart

class VoicesTimeOfDayTextFieldController extends ValueNotifier<TimeOfDay?> {
  VoicesTimeOfDayTextFieldController([super._value]);
}

class VoicesTimeOfDayTextField extends StatefulWidget {
  const VoicesTimeOfDayTextField({super.key});

  @override
  State<VoicesTimeOfDayTextField> createState() =>
      _VoicesTimeOfDayTextFieldState();
}

class _VoicesTimeOfDayTextFieldState extends State<VoicesTimeOfDayTextField> {
  @override
  Widget build(BuildContext context) {
    return VoicesTextField(
      onFieldSubmitted: (value) {},
    );
  }
}

```

Reply:
I'm sure this will be harded to implement then what i just shown so we may discuss this with frontend team before committing to it